LeanOA/ForMathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/Abs.lean

```diff
@@ -0,0 +1,285 @@
+/-
+Copyright (c) 2024 Jon Bannon. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jon Bannon, Jireh Loreaux
+-/
+
+import Mathlib.Analysis.SpecialFunctions.ContinuousFunctionalCalculus.Rpow
+import Mathlib.Analysis.SpecialFunctions.ContinuousFunctionalCalculus.PosPart
+import Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Order
+
+/-!
+# Absolute value of an operator defined via the continuous functional calculus
+
+This file defines the absolute value via the (unital and non unital) continuous functional calculus
+(CFC) and (CFCₙ), and includes foundational API.
+
+## Main declarations
+
++ `CFC.abs`: The absolute value declaration as `abs a := sqrt (star a) * a`.
+
+# TODO
+
+There is likely an `RCLike` version of `abs_smul_complex`.
+
+-/
+
+open scoped NNReal
+
+namespace CFC
+
+section NonUnital
+
+variable {A : Type*} [NonUnitalNormedRing A] [StarRing A]
+
+section abs
+
+variable [PartialOrder A] [Module ℝ≥0 A] [SMulCommClass ℝ≥0 A A] [IsScalarTower ℝ≥0 A A]
+variable [NonUnitalContinuousFunctionalCalculus ℝ≥0 (fun (a : A) => 0 ≤ a)]
```

Comment:
When you need a CFC over a `ℝ≥0`, these are I think the default generic assumptions. 
I don't think you should ever really be assuming you have a CFC over `ℝ≥0` itself. And  working with a CFC for positive elements without `StarOrderedRing` is just weird. I think this will fix most of your problems. Note also, you don't need `Normed` anymore.
```lean
variable {A : Type*} [NonUnitalRing A] [StarRing A] [TopologicalSpace A]
variable [PartialOrder A] [StarOrderedRing A] [Module ℝ A] [SMulCommClass ℝ A A] [IsScalarTower ℝ A A]
variable [NonUnitalContinuousFunctionalCalculus ℝ (IsSelfAdjoint : A → Prop)]
variable [NonnegSpectrumClass ℝ A]
```
You can then edit these hypotheses mildly to get a complex algebra, when you need it, or a unital one.